src/main/java/seedu/address/model/person/Person.java

```diff
@@ -111,6 +117,60 @@ public Address getAddress() {
         return Collections.unmodifiableSet(tags.get().toSet());
     }
 
+    /**
+     * Returns whether a person's name contains some of the specified keywords,
+     * within a Levenshtein distance of {@link #FIND_NAME_DISTANCE_TOLERANCE},
+     * ONLY WHEN the name word contains a number of characters
+     * more than or equal to the global limit {@link #FIND_NAME_GLOBAL_TOLERANCE}.
+     *
+     * Otherwise, it returns the case-insensitive direct match for the particular
+     * name word and continues on to the next name word for a check.
+     *
+     * If no conditions are triggered, the method returns false.
+     *
+     * @param keyWords for searching
+     * @return {@code true} if the name matches or is close to any keyWord,
+     * {@code false} otherwise.
+     */
+    @Override
+    public boolean isNameCloseToAnyKeyword(List<String> keyWords) {
+        final List<String> wordsInName = getName().getWordsInName();
+        for (String nameWord : wordsInName) {
+            if (nameWord.length() >= FIND_NAME_GLOBAL_TOLERANCE) {
+                for (String keyWord : keyWords) {
+                    if (levenshteinDistance(nameWord, keyWord) <= FIND_NAME_DISTANCE_TOLERANCE) {
```

Comment:
can improve SLAP here? the arrow head code is an indicator of bad SLAP

Reply:
Good point. Thanks @okkhoy